source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -162,6 +162,13 @@ void TextureBaker::optimizeBakedTextures(NodePtr shader)
         return;
     }
 
+    // Early exist if not optimizing	
```

Comment:
Fix for when optimize constants is turned off. Used to not add an image input, but also ended up not writing any values as well during baking.

Reply:
I think we still need to check for uniform images when optimizeConstants is set to false, so that we can still return 4x4 textures instead of full-sized textures when individual UDIMs are constant.  The idea behind "optimizeConstants" is that some tools don't support constant values, only images, but there are still optimizations that are valuable for tools that have this limitation.